File: How-Many-Numbers-Are-Smaller-Than-the-Current-Number.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function (nums) {
    let ans = [];
    let originalNums = nums.slice();
    nums.sort((a, b) => a - b);
    for (let i = 0; i < nums.length; i++) {
        let index = nums.indexOf(originalNums[i]);
        ans.push(index);
    }
    return ans;
};

File: How-Many-Numbers-Are-Smaller-Than-the-Current-Number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> ans;
        vector<int> originalNums = nums;
        sort(nums.begin(), nums.end());
        for(int i = 0; i < originalNums.size(); i++) {
            int index = lower_bound(nums.begin(), nums.end(), originalNums[i]) - nums.begin();
            ans.push_back(index);
        }
        return ans;
    }
};